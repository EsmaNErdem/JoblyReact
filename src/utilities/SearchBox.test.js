import { render, fireEvent } from "@testing-library/react";
import SearchBox from "./SearchBox"


test("renders without crashing", async () => {
    render(<SearchBox />); 
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(
        <SearchBox 
            company={true}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});


test("it renders and matches with snaphot", () => {
    const { asFragment } = render(
        <SearchBox 
            job={true}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

test("it renders with company name, min and max employees inputs", () => {
    const { getByText } = render(
        <SearchBox 
            company={true}
        />
    );
    expect(getByText("Name")).toBeInTheDocument();
    expect(getByText("Min Employees")).toBeInTheDocument();
    expect(getByText("Max Employees")).toBeInTheDocument();
});

test("it renders with company name, min and max employees inputs", () => {
    const { getByText } = render(
        <SearchBox 
            job={true}
        />
    );
    expect(getByText("Title")).toBeInTheDocument();
    expect(getByText("Min Salary")).toBeInTheDocument();
    expect(getByText("Has Equity")).toBeInTheDocument();
});


test("it submits form", () => {
    const mockSearch = jest.fn().mockResolvedValue({});
    const { getByLabelText, getByText } = render(
        <SearchBox 
            company={true}
            search={mockSearch}
        />
    );
    const nameInput = getByLabelText("Name");
    const minEmployeeInput = getByLabelText("Min Employees");
    const maxEmployeeInput = getByLabelText("Max Employees");
    const searchButton = getByText("Search")

    fireEvent.change(nameInput, { target: { value: "C1" } });
    fireEvent.change(minEmployeeInput, { target: { value: 100 } });
    fireEvent.change(maxEmployeeInput, { target: { value: 500 } });
    fireEvent.click(searchButton);
  
    expect(mockSearch).toHaveBeenCalledWith({
      name: "C1",
      minEmployees: "100",
      maxEmployees: "500"
    });  
});
