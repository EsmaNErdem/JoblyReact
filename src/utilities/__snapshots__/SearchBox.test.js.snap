// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snaphot 1`] = `
<DocumentFragment>
  <div
    class="SearchBox"
  >
    <form>
      <label
        for="search"
      >
        Name
      </label>
      <input
        id="search"
        name="name"
        placeholder="name"
        type="search"
        value=""
      />
      <label
        for="minEmployees"
      >
        Min Employees
      </label>
      <input
        id="minEmployees"
        min="0"
        name="minEmployees"
        type="number"
        value=""
      />
      <label
        for="maxEmployees"
      >
        Max Employees
      </label>
      <input
        id="maxEmployees"
        name="maxEmployees"
        type="number"
        value=""
      />
      <button>
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`it renders and matches with snaphot 2`] = `
<DocumentFragment>
  <div
    class="SearchBox"
  >
    <form>
      <label
        for="search"
      >
        Title
      </label>
      <input
        id="search"
        name="title"
        placeholder="title"
        type="search"
        value=""
      />
      <label
        for="minSalary"
      >
        Min Salary
      </label>
      <input
        id="minSalary"
        min="0"
        name="minSalary"
        type="number"
        value=""
      />
      <label
        for="hasEquity"
      >
        Has Equity
      </label>
      <input
        id="hasEquity"
        name="hasEquity"
        type="checkbox"
        value="false"
      />
      <button>
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
